VideoLocker/src/main/java/org/edx/mobile/view/CourseCombinedInfoFragment.java

```diff
@@ -345,23 +340,71 @@ private void populateFacePile(){
 
     }
 
-    private View generateAnnouncementView(AnnouncementsModel model){
+    private void populateAnnouncements(List<AnnouncementsModel> announcementsList) {
+        if(announcementsList !=null && announcementsList.size()>0) {
+            hideEmptyAnnouncementMessage();
 
-        View convertView = inflater.inflate(R.layout.row_announcement_list, null);
+            // stop existing views before we remove them from the layout
+            for (AnnouncementView view : announcementViewList) {
+                view.stop();
+            }
+            announcementViewList.clear();
+            announcementContainer.removeAllViews();
+
+            for (AnnouncementsModel m : announcementsList) {
+                AnnouncementView announcementView = generateAnnouncementView(m);
+                announcementViewList.add(announcementView);
+
+                announcementContainer.addView(announcementView.getView());
+            }
+        } else {
+            showEmptyAnnouncementMessage();
+        }
+    }
 
-        TextView date = (TextView) convertView.findViewById(R.id.announcement_date);
-        TextView content = (TextView) convertView.findViewById(R.id.announcement_content);
+    private AnnouncementView generateAnnouncementView(AnnouncementsModel model){
+        AnnouncementView view = new AnnouncementView(inflater, model);
+        view.render();
+        return view;
+    }
 
-        date.setText(model.getDate());
+    private static class AnnouncementView {
+        private View convertView;
+        private TextView date;
+        private WebView webView;
+        private AnnouncementsModel model;
+
+        AnnouncementView(LayoutInflater inflater, AnnouncementsModel model) {
+            this.model = model;
+            this.convertView = inflater.inflate(R.layout.row_announcement_list, null);
+            this.date = (TextView) convertView.findViewById(R.id.announcement_date);
+            this.webView = (WebView) convertView.findViewById(R.id.announcement_content_webview);
+        }
 
-        Spanned text = Html.fromHtml(model.content);
+        void render() {
+            date.setText(model.getDate());
+            webView.loadDataWithBaseURL(Config.getInstance().getApiHostURL(), model.content, "text/html", HTTP.UTF_8, null);
```

Comment:
We probably need to modify the content so it includes ``<meta name="viewport" `` as described here https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag 

Otherwise the text will be incredibly tiny.